clients/src/test/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegateTest.java

```diff
@@ -169,14 +184,63 @@ public void testHasAnyPendingRequests() throws Exception {
         }
     }
 
+    @Test
+    public void testPropagateMetadataError() {
+        String exMsg = "Test Auth Exception";
+        doThrow(new AuthenticationException(exMsg)).when(metadata).maybeThrowAnyException();
+
+        LinkedList<BackgroundEvent> backgroundEventQueue = new LinkedList<>();
+        this.backgroundEventHandler = new BackgroundEventHandler(backgroundEventQueue);
+        NetworkClientDelegate networkClientDelegate = newNetworkClientDelegate();
+
+
+        assertEquals(0, backgroundEventQueue.size());
+        networkClientDelegate.poll(0, time.milliseconds());
+        assertEquals(1, backgroundEventQueue.size());
+
+        ErrorEvent event = (ErrorEvent) backgroundEventQueue.poll();
+        assertNotNull(event);
+        assertEquals(AuthenticationException.class, event.error().getClass());
+        assertEquals(exMsg, event.error().getMessage());
+    }
+
+    @Test
+    public void testPropagateInvalidTopicMetadataError() {
+        LinkedList<BackgroundEvent> backgroundEventQueue = new LinkedList<>();
+        this.backgroundEventHandler = new BackgroundEventHandler(backgroundEventQueue);
+        this.metadata = new Metadata(100, 100, 50000,
```

Comment:
can we also use the mocked metadata as well?

Reply:
There is a suggestion to remove that test, what do you think?
https://github.com/apache/kafka/pull/16043#discussion_r1631395106

Reply:
yeah, we should just remove one.